relentless/optimize/objective.py

```diff
@@ -279,9 +281,35 @@ def compute(self, directory=None):
             The result, which has unknown value ``None`` and known gradient.
 
         """
+        # run simulation and use result to compute gradient
         sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
         sim_ens = self.thermo.extract_ensemble(sim)
+        gradient = self.gradient(sim_ens)
+
+        # optionally write output to directory
+        if directory is not None and (self.communicator is None or self.communicator.rank == self.communicator.root):
+            for n,p in enumerate(self.potentials.pair.potentials):
+                p.save(directory.file('potential.{}.json'.format(n)))
+            sim_ens.save(directory.file('ensemble.json'))
+
+        # relative entropy *value* is None
+        return self.make_result(None, gradient, directory)
+
+    def gradient(self, sim_ens):
```

Comment:
Question: do we want this to be called `gradient` or `compute_gradient`? The former is shorter but also seems more like a property than a method.

In any case, suggesting a nicer argument name:
```suggestion
    def gradient(self, ensemble):
```